cmd/snap-bootstrap/initramfs_systemd_mount_test.go

```diff
@@ -55,6 +55,44 @@ exit 1
 	c.Assert(err, ErrorMatches, "mocked error")
 }
 
+// This is needed to properly split options that contain a ',' character that
+// is double escaped before passed as an argument to systemd-mount.
+func splitOptions(s string) []string {
```

Comment:
Should this live in osutils? We just had a bunch of bugs related to parsing mount options in /proc/PID/mounts or /proc/mounts.

Reply:
So I took a look at the kind of parsers we have in osutil today and I think not for now but could be a candidate in the future. That is because right now I couldn't find any other place where we have to parse `systemd-mount`'s `--options` arguments. What happens here is that `doSystemdMountImpl` is the one constructing the cmdline options and then we need to parse them only for the unit tests' purposes.

For example for the following:
```bash
systemd-mount --options=ro,nosuid,bind,lowerdir=/lower\,,upperdir=/upper
```
The new overlay related arguments complicate this parsing process, which was simply a `strings.Split()` before, as they could contain commas since for instance `lower,` could be a valid directory name.

Reply:
Isn't this exactly option parser we already have to have for parsing options that we read back from `/proc/PID/{mountinfo,mounts}`? I suspect there might be some different escaping in use (because everything is terrible and it's probably not consistent in some way) but I think having it in osutil as a separate piece of code we reach out to is better than having it stashed away in a more obscure places that fewer eyes visit.

Reply:
So I tried the `mountinfo` first (which also splits the options on comma). If I mount an overlay with:
```bash
sudo mount -t overlay overlay merged --options=lowerdir=lower\\,,upperdir=upper,workdir=work
```
It shows in `/proc/self/mountinfo` as:
```
275 31 0:224 / /tmp/merged rw,relatime shared:56 - overlay overlay rw,lowerdir=lower\134\054,upperdir=upper,workdir=work,nouserxattr
```
so in that case split on comma will still work because mountinfo will encode the comma (and the \) in octal.

Then I tried the `mountentry` parser. This seems like a valid fstab entry:
```
overlay /merged overlay noauto,x-systemd.automount,lowerdir=/lower,upperdir=/upper\,,workdir=/work 0 0
```
by verifying it with `findmnt -x`. So I moved this function in there, fixed a bug in it that was discovered from a separate failing test case and also added an extra test case. All tests seem to pass but I would appreciate one more look.